scylla-cql/src/types/serialize/value.rs

```diff
@@ -29,19 +48,940 @@ pub trait SerializeCql {
+    t0, t1, t2, t3, t4, t5, t6, t7, t8, t9, t10, t11, t12, t13, t14, t15;
+    16
+);
+
+fn serialize_sequence<'t, T: SerializeCql + 't, W: CellWriter>(
+    rust_name: &'static str,
+    len: usize,
+    iter: impl Iterator<Item = &'t T>,
+    typ: &ColumnType,
+    writer: W,
+) -> Result<W::WrittenCellProof, SerializationError> {
+    let elt = match typ {
+        ColumnType::List(elt) | ColumnType::Set(elt) => elt,
+        _ => {
+            return Err(mk_typck_err_named(
+                rust_name,
+                typ,
+                SetOrListTypeCheckErrorKind::NotSetOrList,
+            ));
+        }
+    };
+
+    let mut builder = writer.into_value_builder();
+
+    let element_count: i32 = len.try_into().map_err(|_| {
+        mk_ser_err_named(
+            rust_name,
+            typ,
+            SetOrListSerializationErrorKind::TooManyElements,
+        )
+    })?;
+    builder.append_bytes(&element_count.to_be_bytes());
+
+    for el in iter {
+        T::serialize(el, elt, builder.make_sub_writer()).map_err(|err| {
+            mk_ser_err_named(
+                rust_name,
+                typ,
+                SetOrListSerializationErrorKind::ElementSerializationFailed(err),
+            )
+        })?;
+    }
+
+    builder
+        .finish()
+        .map_err(|_| mk_ser_err_named(rust_name, typ, BuiltinSerializationErrorKind::SizeOverflow))
+}
+
+fn serialize_mapping<'t, K: SerializeCql + 't, V: SerializeCql + 't, W: CellWriter>(
+    rust_name: &'static str,
+    len: usize,
+    iter: impl Iterator<Item = (&'t K, &'t V)>,
+    typ: &ColumnType,
+    writer: W,
+) -> Result<W::WrittenCellProof, SerializationError> {
+    let (ktyp, vtyp) = match typ {
+        ColumnType::Map(k, v) => (k, v),
+        _ => {
+            return Err(mk_typck_err_named(
+                rust_name,
+                typ,
+                MapTypeCheckErrorKind::NotMap,
+            ));
+        }
+    };
+
+    let mut builder = writer.into_value_builder();
+
+    let element_count: i32 = len.try_into().map_err(|_| {
+        mk_ser_err_named(rust_name, typ, MapSerializationErrorKind::TooManyElements)
+    })?;
+    builder.append_bytes(&element_count.to_be_bytes());
+
+    for (k, v) in iter {
+        K::serialize(k, ktyp, builder.make_sub_writer()).map_err(|err| {
+            mk_ser_err_named(
+                rust_name,
+                typ,
+                MapSerializationErrorKind::KeySerializationFailed(err),
+            )
+        })?;
+        V::serialize(v, vtyp, builder.make_sub_writer()).map_err(|err| {
+            mk_ser_err_named(
+                rust_name,
+                typ,
+                MapSerializationErrorKind::ValueSerializationFailed(err),
+            )
+        })?;
+    }
+
+    builder
+        .finish()
+        .map_err(|_| mk_ser_err_named(rust_name, typ, BuiltinSerializationErrorKind::SizeOverflow))
+}
 
 pub fn serialize_legacy_value<T: Value, W: CellWriter>(
```

Comment:
Is this used anywhere after the changes in this PR? I can't find any references to it. If not, it can be removed